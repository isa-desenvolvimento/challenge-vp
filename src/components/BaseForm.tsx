import { Formik, Form } from 'formik'

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const BaseForm = ({ initialValues, baseForm, onSubmit }: any) => {
  const renderForm = () => {
    return (
      <Formik
        onSubmit={(data, action) => onSubmit(data, action)}
        initialValues={initialValues}
      >
        {() => <Form>{baseForm}</Form>}
      </Formik>
    )
  }

  return <>{renderForm()}</>
}
