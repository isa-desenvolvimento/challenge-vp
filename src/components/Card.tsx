import {
  StyleCardContainer,
  StyleCardContent,
  StyleCardFooter,
  StyleCardHeader,
} from '@/styles/Card'
import { CardProps } from '@/types/global'
import React from 'react'

export const Card = ({
  header,
  content,
  footer,
  primary,
  display,
  height,
}: CardProps) => {
  const renderHeader = () =>
    header && <StyleCardHeader primary={primary}>{header}</StyleCardHeader>
  const renderContent = () =>
    content && (
      <StyleCardContent primary={primary} height={height}>
        {content}
      </StyleCardContent>
    )
  const renderFooter = () =>
    footer && <StyleCardFooter primary={primary}>{footer}</StyleCardFooter>

  return (
    <StyleCardContainer display={display}>
      {renderHeader()}
      {renderContent()}
      {renderFooter()}
    </StyleCardContainer>
  )
}
