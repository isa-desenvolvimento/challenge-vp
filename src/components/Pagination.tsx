import {
  StylePaginationContainer,
  StylePaginationNumber,
  StylePaginationContent,
} from '@/styles/Pagination'
import { PaginationProps } from '@/types/global'

export const Pagination = ({ total }: PaginationProps) => {
  return (
    <StylePaginationContainer role="pagination">
      <StylePaginationContent>
        <StylePaginationNumber>{` Total ${total} registros`}</StylePaginationNumber>
      </StylePaginationContent>
    </StylePaginationContainer>
  )
}
