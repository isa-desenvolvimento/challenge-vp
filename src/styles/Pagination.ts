import styled from 'styled-components'

export const StylePaginationContainer = styled.div`
  display: grid;
  display: grid;
  grid-auto-columns: 80% auto;
  text-align: justify;

  @media screen and (max-width: 600px) {
    display: revert;
    width: 100%;
  }
`

export const StylePaginationNumber = styled.div`
  background: white;
  border-radius: 4px;
  padding: 0.4rem;
  text-align: center;

  align-content: center;
  justify-content: space-between;
`

export const StylePaginationContent = styled.div`
  display: flex;
  text-align: justify;
  gap: 1rem;
`
