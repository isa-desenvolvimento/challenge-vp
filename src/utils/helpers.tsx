import { render } from '@testing-library/react'
import GlobalContext from '../context'

export const renderWithProvider = (children: React.ReactNode) => {
  render(<GlobalContext>{children}</GlobalContext>)
}
