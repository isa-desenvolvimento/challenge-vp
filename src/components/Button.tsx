import { Icon } from '@/components/Icon'
import { StyleButton } from '@/styles/Button'
import { ButtonProps } from '@/types/global'

export const Button = ({
  iconProps,
  primary,
  onClick,
  children,
  size,
  type,
}: ButtonProps) => {
  const renderIcon = () => iconProps && <Icon faIcon={iconProps}></Icon>

  return (
    <StyleButton
      role="button"
      arial-label="button"
      primary={primary}
      size={size}
      type={type}
      onClick={onClick}
    >
      {renderIcon()}
      {children}
    </StyleButton>
  )
}
