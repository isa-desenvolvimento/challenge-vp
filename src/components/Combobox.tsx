import { ComboboxProps } from '@/types/global'

import { DropdownWrapper, StyledSelect, StyledOption } from '@/styles/Combobox'
import { useField } from 'formik'

export const Combobox = ({
  role,
  options,
  placeholder,
  primary,
  onChange,
  name,
  isFormik,
}: ComboboxProps) => {
  const [field, meta] = useField(name)

  const renderSelection = () => {
    if (isFormik) {
      return (
        <StyledSelect primary={primary} {...field} {...meta} role={role}>
          <StyledOption selected value="" name={`${name}.defaultValue`}>
            {placeholder}
          </StyledOption>
          {renderOptions()}
        </StyledSelect>
      )
    }

    return (
      <StyledSelect onChange={onChange} primary={primary}>
        <StyledOption selected value="" name={`${name}.defaultValue`}>
          {placeholder}
        </StyledOption>
        {renderOptions()}
      </StyledSelect>
    )
  }

  const renderOptions = () => {
    return (
      options?.length &&
      options.map(({ value }) => {
        return (
          <StyledOption name={`${name}.${value}`} key={value} value={value}>
            {value}
          </StyledOption>
        )
      })
    )
  }

  return <DropdownWrapper role="combobox">{renderSelection()}</DropdownWrapper>
}
