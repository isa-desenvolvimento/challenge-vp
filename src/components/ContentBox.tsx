import { StyleContainer } from '@/styles/ContentBox'
import { ChildrenProps } from '@/types/global'

export const ContentBox = ({ children }: ChildrenProps) => {
  return <StyleContainer role="contentBox"> {children}</StyleContainer>
}
