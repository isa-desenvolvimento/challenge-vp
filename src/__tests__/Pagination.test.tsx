import { renderWithProvider } from '../utils/helpers'
import { screen } from '@testing-library/react'

import { Pagination } from '@/components/Pagination'

describe('<Pagination />', () => {
  beforeEach(() =>
    renderWithProvider(<Pagination prev={4} next={2} total={12} />)
  )

  it('should render text pagination', () => {
    expect(screen.getByRole(/pagination/i)).toBeTruthy()
  })

  // it('should render the styled-components pagination', () => {
  //   expect(screen.getByRole(/pagination/i)).toHaveStyle({
  //     background: '#36a4a8',
  //     color: '#ffff',
  //     'font-size': '1em',
  //     margin: '1em',
  //     border: 'none',
  //     'border-radius': '4px',
  //     padding: '0.8em 2em',
  //     width: '10%',
  //     'text-transform': 'uppercase',
  //     'text-overflow': 'ellipsis',
  //     cursor: 'pointer',
  //   })
  // })

  it('should render snapshot', () => {
    expect(screen.getByRole(/pagination/i)).toMatchSnapshot()
  })
})
