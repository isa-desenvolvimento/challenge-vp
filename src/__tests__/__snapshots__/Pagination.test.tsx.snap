// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> should render snapshot 1`] = `
.c0 {
  display: grid;
  display: grid;
  grid-auto-columns: 80% auto;
  text-align: justify;
}

.c2 {
  background: white;
  border-radius: 4px;
  padding: 0.4rem;
  text-align: center;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  text-align: justify;
  gap: 1rem;
}

@media screen and (max-width:600px) {
  .c0 {
    display: revert;
    width: 100%;
  }
}

<div
  class="c0"
  role="pagination"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
       Total 12 registros
    </div>
  </div>
</div>
`;
